src/entry.rs

```diff
@@ -56,9 +60,47 @@ impl Entry {
         let entry = Self::new(start_hash, *cur_hashes, transactions);
         *start_hash = entry.id;
         *cur_hashes = 0;
+        assert!(serialized_size(&entry).unwrap() <= BLOB_DATA_SIZE as u64);
         entry
     }
 
+    /// Creates the next entries for given transactions, outputs
+    /// updates start_hash to id of last Entry, sets cur_hashes to 0
+    pub fn from_transactions_mut(
```

Comment:
Seems like this should be in ledger.rs, where Block is defined.

Reply:
agreed, refactoring